### PR TITLE
Add attr_json plugin

### PR DESCRIPTION
**Pull request by caiofilipemr**
Add plugin to [attr_json](https://github.com/jrochkind/attr_json) gem

Tested on our private repository.

### CODE REVIEW

File: lib/sorbet-rails/gem_plugins/attr_json_plugin.rb [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    custom_module_rbi.create_method(
+        "#{definition.name}=",
+        parameters: [
+          ::Parlour::RbiGenerator::Parameter.new(
+            'value',
+            type: 'T.untyped'
+          )
+        ],
+        returns: 'T.untyped'
```
Comment (manhhung741) [inline lib/sorbet-rails/gem_plugins/attr_json_plugin.rb:87]: nit: keep using 2 space indentation

File: lib/sorbet-rails/gem_plugins/attr_json_plugin.rb [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    )
+
+    custom_module_rbi.create_method(definition.name.to_s,
+                                    returns: definition_type)
```
Comment (manhhung741) [inline lib/sorbet-rails/gem_plugins/attr_json_plugin.rb:91]: nit: to keep the styling consistent, can you update the style following this suggestion:
```suggestion
    custom_module_rbi.create_method(
      definition.name.to_s,
      returns: definition_type,
    )
```

File: lib/sorbet-rails/gem_plugins/attr_json_plugin.rb [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    custom_module_rbi.create_method("#{definition.name}?",
+                                    returns: 'T::Boolean')
```
Comment (manhhung741) [inline lib/sorbet-rails/gem_plugins/attr_json_plugin.rb:94]: ditto

File: lib/sorbet-rails/gem_plugins/attr_json_plugin.rb [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      "T.nilable(#{type_to_class(definition.type.type.to_s)})"
+    end
+  end
+
+  sig { params(type: String).returns(String) }
+  def type_to_class(type)
+    return 'T.untyped' unless type.present?
+
+    case type
+    when 'datetime' then 'DateTime'
```
Comment (manhhung741) [inline lib/sorbet-rails/gem_plugins/attr_json_plugin.rb:112]: Usually, datetime type will map to `ActiveSupport::TimeWithZone`. Can you double-check?

File: lib/sorbet-rails/gem_plugins/attr_json_plugin.rb (hunk 1 above)
Comment (manhhung741) [inline lib/sorbet-rails/gem_plugins/attr_json_plugin.rb:87]: I saw you intentionally changed the types of `value` & the method return type to `T.untyped`. Can you explain (and add a comment in the code) why it didn't work using "definition_type" (or "void" for the method return type)?

**Review by manhhung741 — CHANGES_REQUESTED** [review]
Looking pretty solid. I had some comments about the code styling and request additional comments added to the code.

**Comment by manhhung741** [comment]
Thank you @caiofilipemr for adding the plugins! I super appreciate it! 🚀

File: lib/sorbet-rails/gem_plugins/attr_json_plugin.rb (hunk 4 above)
Comment (caiofilipemr) [reply lib/sorbet-rails/gem_plugins/attr_json_plugin.rb:112]: Yeah, it does not map to `DateTime` but to `Time`:

```ruby
> Post.first.my_datetime.class
=> Time
```

I will fix. Thank you.

File: lib/sorbet-rails/gem_plugins/attr_json_plugin.rb (hunk 1 above)
Comment (caiofilipemr) [reply lib/sorbet-rails/gem_plugins/attr_json_plugin.rb]: Added [a comment](https://github.com/chanzuckerberg/sorbet-rails/blob/38e3ffb81a62a1f7542358fdccf8ceced054269c/lib/sorbet-rails/gem_plugins/attr_json_plugin.rb#L77-L91). Hope it clarified, if not I can improve it :smile:

**Comment by caiofilipemr** [comment]
Hi, @manhhung741, do you need me to change anything else here too?

**Review by manhhung741 — APPROVED** [review]
Thanks for updating the PR & sorry for the late review.
Can you rebase the changes so that I can merge them? Ideally, if you can give me edit permissions I can help rebasing them. (I'm not sure if the permission can be changed now)

**Comment by manhhung741** [comment]
Ah. You can follow this instruction to allow a maintainer to update the PR: https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by caiofilipemr** [comment]
> Can you rebase the changes so that I can merge them? Ideally, if you can give me edit permissions I can help rebasing them. (I'm not sure if the permission can be changed now)

Thank you for the reply!

I couldn't find this option, maybe I don't have permission because I'm not the repository owner, Idk. So, I rebase myself, I hope it is ok!

**Pull request merged**